src/main/scala/org/openeo/opensearch/OpenSearchResponses.scala

```diff
@@ -270,11 +265,27 @@ object OpenSearchResponses {
     }
 
     val featuresGroupedFiltered = dupClusters.map({ case (key, features) =>
-      val selectedElement = features.maxBy(_.generalProperties.published)
-      (key, selectedElement)
+      val selectedElement = features.maxBy(_.deduplicationOrderValue)
+      val toBeRemoved = features.filter(_ != selectedElement)
+      val toLog = if (toBeRemoved.nonEmpty)
+        ("Removing duplicated feature(s): "
+          + toBeRemoved.map(f => "id=" + f.id + "' deduplicationOrderValue='"
+          + f.deduplicationOrderValue.getOrElse("") + "'").mkString(", ")
+          + ". Keeping the Latest published one: id=" + selectedElement.id + "' deduplicationOrderValue='"
```

Comment:
still a reference to 'latest published', while it could not be ordered by something else?

Reply:
Thanks! Fixed